cypress-tests/cypress/e2e/spec/Payment/00031-UCSFlowTest.cy.js

```diff
@@ -0,0 +1,184 @@
+        );
+        cy.task(
+          "cli_log",
+          `📝 Testable requests: ${testableRequests.join(", ")}`
+        );
+        cy.task(
+          "cli_log",
+          `📈 Test Coverage: ${testableRequests.length}/${allRequests.length} (${((testableRequests.length / allRequests.length) * 100).toFixed(1)}%)`
+        );
+
+        if (testableRequests.length === 0) {
+          throw new Error(
+            `No UCS-compatible requests found for connector: ${currentConnector}`
+          );
+        }
+      });
+
+      it("should setup UCS environment", () => {
+        cy.setupUCSEnvironment(
+          { ...fixtures, payment_methods_enabled },
+          globalState,
+          currentConnector
+        );
+        if (shouldContinue) shouldContinue = true;
+      });
+
+      // Dynamic test generation - following pattern from other test files
+      UCS_REQUEST_NAMES.forEach((requestType) => {
+        it(`should test ${requestType}`, () => {
+          // Skip test if request not available for this connector
+          if (!testableRequests.includes(requestType)) {
+            cy.task(
+              "cli_log",
+              `Skipping ${requestType} - not available for ${currentConnector}`
+            );
+            return;
+          }
+
+          const data = getConnectorDetails(globalState.get("connectorId"))[
+            "card_pm"
+          ][requestType];
+
+          if (!data?.Request || !data?.Response) {
+            throw new Error(`Configuration missing for: ${requestType}`);
+          }
+
+          cy.task("cli_log", `Testing ${requestType}`);
+
+          if (requestType.includes("MIT") || requestType.includes("Repeat")) {
+            cy.task(
+              "cli_log",
+              `Skipping ${requestType} - requires existing mandate setup`
+            );
+            return;
+          }
+
+          if (requestType === "UCSZeroAuthMandate") {
+            // Get full connector config for sequential flow
+            const connectorConfig = {
+              card_pm: getConnectorDetails(currentConnector).card_pm,
+            };
+            cy.executeUCSSequentialFlow(
+              connectorConfig,
+              {},
+              currentConnector,
+              globalState
+            );
+          } else {
+            // Individual UCS request test
+            cy.createUCSPayment(
+              requestType,
+              currentConnector,
+              globalState,
+              data.Request
+            ).then((response) => {
+              cy.validateUCSResponse(response, data.Response, requestType).then(
+                (result) => {
+                  if (!result.success) {
+                    shouldContinue = false;
+                    throw new Error(`UCS Test Failed - ${result.error}`);
+                  }
+                }
+              );
+            });
+          }
+
+          // Use standard pattern for continuation
```

Comment:
can you please verify the code generated by the clanker before marking it for review.

this looks like a mess at this point. please have a look at the readme.md. it is not up-to-date, but yeah, it covers beginning part (basics) at least.